src/filerepository.js

```diff
@@ -174,10 +176,20 @@ export default class FileRepository extends Plugin {
 			return null;
 		}
 
-		const loader = new FileLoader( file );
+		const filePromise = fileOrPromise instanceof Promise ? fileOrPromise : new Promise( resolve => resolve( fileOrPromise ) );
```

Comment:
I'd shorten that using chainability of promises
```js
const loader = new FileLoader( Promise.resolve( fileOrPromise ) );
```

If value is returned it will be wrapped in promise. If you return a promise then it will be simply returned and work as intended.